README.md

```diff
@@ -2,6 +2,106 @@
 
 `rdf-cube-view-query` provides an API and query builder to create and query views based on the [cube view schema](https://github.com/zazuko/rdf-cube-view-schema).
 
+## API
+
+The API of this package is built on top of the [rdf cube view schema](https://github.com/zazuko/rdf-cube-view-schema).
+For a better understanding it is worth having a look at the docs of the schema repo.
+Behind all objects there is a shared RDF/JS `Dataset`.
+That requires to keep parent/child relations and calling the `.clear()` method if objects are used multiple times with different children.
+All objects have a `parent` argument in the constructor to keep track of the reference.
+
+### Source
+
+A `Source` defines a SPARQL endpoint and optional a named graph.
+`Source`s are used by the `View` to read the observation data.
+There are two subtypes of `Source`.
+A plain `Source` can't be used in a `View`.
+One of the subtypes must be used.
+
+Supported constructor arguments:
+
+- `endpointUrl`: SPARQL endpoint URL as string, RDF/JS `NamedNode` or Clownface object.
+- `user`: User for the SPARQL endpoint as string.
+- `password`: Password for the SPARQL endpoint as string.
+- `sourceGraph`: Named Graph as string, RDF/JS `NamedNode` or Clownface object.
```

Comment:
Is it possible to also query the union graph? Is this argument optional?

Reply:
It says so here https://github.com/zazuko/rdf-cube-view-query/pull/21/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R15 i see. Lets repeat it hear for clarity?

Reply:
Also could multiple graphs be provided? I don't have a use-case now but could become a thing when we have versions.

Reply:
I didn't add any optional info for the arguments at all. Will do it in a separate PR more focused on code. `undefined` or `null` means union graph, like in any other RDF/JS library. Multiple graphs are not supported. The graph is directly translated to a SPARQL query. Having multiple graphs would make it a little bit complicated.

Reply:
please merge